shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -61,6 +69,108 @@
           .hasArg(false)
           .desc("copy files in subdirectories recursively")
           .build();
+  private static final Option THREAD_OPTION =
+      Option.builder("t")
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(1)
+          .argName("threads")
+          .type(Integer.class)
+          .desc("Number of threads used to copy files in parallel")
+          .build();
+
+  /**
+   * A thread pool for asynchronous copy.
+   *
+   * Copy tasks can send messages to an output stream in a thread safe way.
+   */
+  @ThreadSafe
+  private final class CopyThreadPool {
```

Comment:
the `submit` and `shutdown` are more like an `Executor` than a threadpool, how about `CopyThreadExecutor`?

Reply:
sounds good